pkg/storaged/clevis-dialogs.jsx

```diff
@@ -0,0 +1,199 @@
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import cockpit from "cockpit";
+import dialog from "./dialog";
+import $ from "jquery";
+
+var _ = cockpit.gettext;
+
+export function add(client, block) {
+    dialog.open({ Title: _("Add network key"),
+                  Fields: [
+                      { TextInput: "url",
+                        Title: _("Key server address")
+                      },
+                      { PassInput: "passphrase",
+                        Title: _("Existing passphrase")
+                      }
+                  ],
+                  Action: {
+                      Title: _("Add"),
+                      action: function (vals) {
+                          return client.clevis_overlay.get_adv(vals.url).then(function (info) {
+                              add_adv(client, block, vals.url, info, vals.passphrase);
+                          });
+                      }
+                  }
+    });
+}
+
+function add_adv(client, block, url, info, passphrase) {
+    verify_adv(url, info,
+               _("Verify Key"),
+               null,
+               _("Trust Key"),
+               function () {
+                   return client.clevis_overlay.add(block, url, info.adv, passphrase)
+                                .fail(function (err) {
+                                    console.log("FAIL", err);
+                                });
+               });
+}
+
+function verify_adv( url, info, title, extra, action_title, action) {
+    var port_pos = url.lastIndexOf(":");
+    var host = (port_pos >= 0) ? url.substr(0, port_pos) : url;
+    var port = (port_pos >= 0) ? url.substr(port_pos + 1) : "";
+    var cmd = cockpit.format("ssh $0 tang-show-keys $1", host, port);
+
+    dialog.open({ Title: title,
+                  Fields: [ ],
+                  Body: (
+                      $('<div>').append(
+                          extra ? $('<p>').text(extra) : null,
+                          $('<p>').append(
+                              $('<span>').text(_("Manually verify the key on the server: ")),
+                              $('<pre>').text(cmd)),
+                          $('<p>').append(
+                              $('<span>').text(_("The output should match this text: ")),
+                              (info.sigkeys.length == 1 ?
+                               $('<samp>').text(info.sigkeys[0]) :
+                               $('<pre>').text(info.sigkeys.join("\n"))))
```

Comment:
It seems a bit odd to format one or multiple keys differently. Would be simpler to always call `join()` (that DTRT for one-element arrays) and use `white-space: pre-wrap`?

Reply:
`samp` is inline, while `pre` is a block. @garrett?

Reply:
With one signing key:
![screenshot from 2018-03-26 14-23-20](https://user-images.githubusercontent.com/3228183/37903624-446568ea-3101-11e8-80fc-57ca1cf8848f.png)

With two:
![screenshot from 2018-03-26 14-24-20](https://user-images.githubusercontent.com/3228183/37903665-6552d3c6-3101-11e8-9520-1fcba9ce3cf6.png)

Reply:
Sorry for the delay; I wrote this earlier (more or less) and GitHub didn't save this comment:

Yep:
- `pre` is block by default
- `samp` is inline by default (for computer output)

Related:
- `kbd` is inline for typed input to the computer
- `code` is for code snippets

Notes:

- `samp`, `kbd`, and `code` are all inline and often should be wrapped in a `pre`
  - `pre` just means prefomatted text, whereas the rest of the tags mentioned here all provide semantic meaning (and might be wrapped in `pre`, but not necessarily)
- `kbd` and `samp` can be nested in each other
- `kbd` inside of `kbd` signifies a key to type; example: `<kbd><kbd>Ctrl</kbd> + <kbd>z</kbd></kbd>`

You can think of `samp` and `kbd` as using single backticks inline in Markdown, and `pre` as the triple-backticks for preformatted samples (and `<pre><code></code></pre>` for code samples).

You might want to also use `kbd` for the string that should be typed.

---

**TL;DR**: Correct. :+1: Consider using `kbd` too.

Reply:
@garrett, whoa, thanks!  Could you make the call here?  Is the current code okay, or should I change something?

Reply:
> Consider using `kbd`

... for text that should be typed.

```html
<pre><kbd>ssh ...</kbd></pre>

<pre><samp>mGSngj...</samp></pre>
```

The `<pre>` is there because these are blocks. If you had the text inline, you wouldn't have the `pre`, but would have the `kbd` & `samp` still.

If you have time to do a quick fix in this PR, that would be great. Pushing this to another PR afterward is fine too, as long as it's fixed.

Reply:
Looks like this now:
![screenshot from 2018-03-27 11-59-32](https://user-images.githubusercontent.com/3228183/37957336-598f2408-31b6-11e8-92fc-e6e6e32e925e.png)

Reply:
Oh. Well, I guess it's styled up in PatternFly or within Cockpit CSS.

Drop the `<kbd>` for now, and I guess we should look into fixing it at a more general level first.

Reply:
The gray-on-black colors come from Patternfly.

Reply:
Done.

Reply:
Opened issues @ https://github.com/cockpit-project/cockpit/issues/8904 & https://github.com/patternfly/patternfly/issues/1002